packages/py-moose-lib/moose_lib/streaming/streaming_function_runner.py

```
@@ -99,6 +146,65 @@ def topic_name_to_stream_name(self) -> str:
+        )
+        sys.exit(1)
+
+    # get the flow definition
+    streaming_function_def = getattr(module, streaming_functions[0])
+
+    # get the run function
+    streaming_function_run = streaming_function_def.run
+
+    # get run input type that doesn't rely on the name of the input parameter
+    run_input_type = streaming_function_run.__annotations__[
+        list(streaming_function_run.__annotations__.keys())[0]
```

Ruff (0.14.13)

Prefer `next(...)` over single element slice

Replace with `next(...)`

(RUF015)